src/_pytest/logging.py

```diff
@@ -420,6 +420,54 @@ def __init__(self, config):
 
         self.log_cli_handler = None
 
+        self._setup_cli_logging()
+
+    def _setup_cli_logging(self):
+        terminal_reporter = self._config.pluginmanager.get_plugin("terminalreporter")
+        # FIXME don't set verbosity level and derived attributes of
+        # terminalwriter directly
+        terminal_reporter.verbosity = self._config.option.verbose
```

Comment:
What if we change `verbosity`, `showheader`, etc, to read-only properties? As we have seen, setting them manually is error prone and leads to surprising results.

Reply:
thats a long term goal, a initial gloss over review of terminalwriter indicated, that it is inconsistent in terms of option/configuration usage

Reply:
Why you mean a long term goal @RonnyPfannschmidt ? Changing them to read-only properties should be really simple and would avoid the hack that @thisch had to put in place. 😁

Reply:
Note that this hack does not fully solve all issues with the unit tests. I can't really explain why the terminalwrapper plugin seems to be not ready when LogginPlugin.__init__ is called in the travis-ci jobs (Nonetype has not attribute ...; see https://travis-ci.org/pytest-dev/pytest/jobs/489189341) 

Any ideas?

Reply:
Oh `xdist` disables the terminal writer, so it never gets loaded.

Reply:
@nicoddemus since there seems tobe inconsistent usage, that issue should be resolved in a own pr instead of shoehorning it into this one which is different